scrapy/pipelines/files.py

```diff
@@ -334,6 +334,9 @@ def _warn():
 
         media_guid = hashlib.sha1(to_bytes(url)).hexdigest()  # change to request.url after deprecation
         media_ext = os.path.splitext(url)[1]  # change to request.url after deprecation
+        if not media_ext or not all([char.isalpha() or char.isdigit() for char in media_ext[1:]]):
```

Comment:
It's there to preserve existing functionality, including this test:
```python
>>>self.assertEqual(file_path(Request("http://www.dfsonline.co.uk/get_prod_image.php?img=status_0907_mdm.jpg")),
'full/4507be485f38b0da8a0be9eb2e1dfab8a19223f2.jpg')
```
This seems like a legitimate use case, so we keep the old routine to accommodate it.  We only fall through to the new routine if the resulting file extension contains non alpha/digit characters, indicating it's probably a querystring fragment, not a proper file extension.